go/cli/mcap/cmd/doctor.go

```diff
@@ -55,6 +58,14 @@ func (doctor *mcapDoctor) fatalf(format string, v ...any) {
 	os.Exit(1)
 }
 
+func (doctor *mcapDoctor) exit() {
+	if doctor.errorCount == 0 {
+		os.Exit(0)
+	}
+	color.Red("Encountered %d errors", doctor.errorCount)
+	os.Exit(1)
+}
```

Comment:
I'm alright with this if it seems like the cleanest way to go for now. A couple notes FYI though -
* There's a function defined called `die` which logs an error to Stderr and then calls exit(1). Not sure if we can sensibly reuse that here given doctor using colors in contrast to the rest of the tool.
* Both this function and `die` are affected by a potential problem - they prevent any deferred functions from getting called. This may or may not cause issues in our usage, but it's something I've been meaning to take a closer look at.